nomad/structs/services.go

```diff
@@ -465,6 +470,9 @@ func (sc *ServiceCheck) Hash(serviceID string) string {
 	// use name "true" to maintain ID stability
 	hashBool(h, sc.TLSSkipVerify, "true")
 
+	// Only include TLSServerName if set.
+	hashStringIfNonEmpty(h, sc.TLSServerName)
```

Comment:
Honestly not sure what the comment for this line should be. Let me know if this needs to change.